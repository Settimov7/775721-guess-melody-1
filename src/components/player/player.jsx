import * as React from 'react';
import * as PropTypes from "prop-types";

export const Player = (props) => {
  const {audio, isPlaying, isLoading, onPlayerButtonClick} = props;

  return (
    <div className="game__track">
      <button
        className={`track__button track__button--${ isPlaying ? `pause` : `play`}`}
        type="button"
        disabled={isLoading}
        onClick={onPlayerButtonClick}
      />
      <div className="track__status">
        {audio}
      </div>
    </div>
  );
};

Player.propTypes = {
  audio: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  isPlaying: PropTypes.bool,
  onPlayerButtonClick: PropTypes.func,
};
