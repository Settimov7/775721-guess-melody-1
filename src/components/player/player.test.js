import * as React from 'react';
import renderer from 'react-test-renderer';

import {Player} from "./player";

it(`Player correctly renders`, () => {
  const props = {
    src: `path.mp3`,
  };

  const createNodeMock = () => {
    return {
      focus() {
        return true;
      }
    };
  };

  const tree = renderer.create(<Player{...props} />, {createNodeMock}).toJSON();

  expect(tree).toMatchSnapshot();
});
