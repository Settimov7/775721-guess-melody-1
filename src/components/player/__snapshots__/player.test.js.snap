// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player correctly renders 1`] = `
<div
  className="game__track"
>
  <button
    className="track__button track__button--play"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
</div>
`;
